lib/fshash/walker.go

```diff
@@ -119,5 +164,17 @@ func FillBucket(srcBasePath, destBasePath string, bucket Bucket, hasherFactory f
 			// we could add a hash of inodes to bucket to address this.
 		}
 		return nil
-	})
+	}
+	postVisit := func(node treewalk.Node) error {
+		filenode := node.(*fileWalkNode)
+		filenode.children = nil
+		if filenode.info.IsDir() && destBasePath != "" {
+			if err := fspatch.UtimesNano(filepath.Join(destBasePath, filenode.path), []syscall.Timespec{def.SomewhenTimespec, syscall.NsecToTimespec(filenode.info.ModTime().UnixNano())}); err != nil {
```

Comment:
Please create the struct on a separate line, and add a comment to its creation and its pass to `UtimesNano`. This statement is waaaay too dense :(

You're calling a syscall to set the modtime of the folder after walking its children, correct?